In [1]:
# Import necessary libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set

import warnings
warnings.filterwarnings('ignore')

In [2]:
# Define a function to load in the dataset

filename = 'API_19_DS2_en_csv_v2_4151283.csv'

# Load the data in its' natural format
def read_file_main(filename):
    #colname = 'Country Code'
    df = pd.read_csv(f'{filename}', skiprows=4,)
    #cols = ['Country Code', 'Indicator Code', 'Indicator Name']
    #df = df.drop(cols, axis=1)
    return df

# Load the data by droping unwanted columns and outputing only needed columns
def read_file(filename):
    #filename = 'API_19_DS2_en_csv_v2_4151283.csv'
    #colname = 'Country Code'
    df1 = pd.read_csv(f'{filename}', skiprows=4,)
    cols = ['Country Code', 'Indicator Code', 'Indicator Name']
    df1 = df1.drop(cols, axis=1)
    return df1

# Load the data in a pivoted order
def read_file_transposed(filename):
    #filename = 'API_19_DS2_en_csv_v2_4151283.csv'
    #colname = 'Country Code'
    df2 = pd.read_csv(f'{filename}', skiprows=4,)
    #cols = ['Country Code', 'Indicator Code', 'Indicator Name']
    df2 = pd.pivot_table(data, values=[col for col in data if col.startswith('1') | col.startswith('2')], columns=data['Country Name'])
    df2 = df2.rename_axis('Year')
    return df2

data = read_file_main(filename)
data1 = read_file(filename)
data2 = read_file_transposed(filename)

In [3]:
# First five records

data.head()

Country Name Country Code                            Indicator Name  \
0        Aruba          ABW  Urban population (% of total population)   
1        Aruba          ABW                          Urban population   
2        Aruba          ABW        Urban population growth (annual %)   
3        Aruba          ABW                         Population, total   
4        Aruba          ABW              Population growth (annual %)   

      Indicator Code       1960          1961          1962          1963  \
0  SP.URB.TOTL.IN.ZS     50.776     50.761000     50.746000     50.730000   
1        SP.URB.TOTL  27525.000  28139.000000  28537.000000  28763.000000   
2        SP.URB.GROW        NaN      2.206183      1.404498      0.788835   
3        SP.POP.TOTL  54208.000  55434.000000  56234.000000  56699.000000   
4        SP.POP.GROW        NaN      2.236462      1.432843      0.823502   

           1964          1965  ...           2013           2014  \
0     50.715000     50.700000  ...      42.990000      43.041000   
1  28922.000000  29080.000000  ...   44351.000000   44666.000000   
2      0.551271      0.544810  ...       0.660561       0.707733   
3  57029.000000  57357.000000  ...  103165.000000  103776.000000   
4      0.580334      0.573498  ...       0.583290       0.590508   

            2015           2016           2017           2018           2019  \
0      43.108000      43.192000      43.293000      43.411000      43.546000   
1   44978.000000   45293.000000   45614.000000   45949.000000   46294.000000   
2       0.696090       0.697901       0.706219       0.731740       0.748028   
3  104339.000000  104865.000000  105361.000000  105846.000000  106310.000000   
4       0.541048       0.502860       0.471874       0.459266       0.437415   

            2020  2021  Unnamed: 66  
0      43.697000   NaN          NaN  
1   46654.000000   NaN          NaN  
2       0.774631   NaN          NaN  
3  106766.000000   NaN          NaN  
4       0.428017   NaN          NaN  

[5 rows x 67 columns]

In [4]:
data1.head()

Country Name       1960          1961          1962          1963  \
0        Aruba     50.776     50.761000     50.746000     50.730000   
1        Aruba  27525.000  28139.000000  28537.000000  28763.000000   
2        Aruba        NaN      2.206183      1.404498      0.788835   
3        Aruba  54208.000  55434.000000  56234.000000  56699.000000   
4        Aruba        NaN      2.236462      1.432843      0.823502   

           1964          1965          1966          1967          1968  ...  \
0     50.715000     50.700000     50.685000     50.670000     50.654000  ...   
1  28922.000000  29080.000000  29246.000000  29411.000000  29570.000000  ...   
2      0.551271      0.544810      0.569216      0.562594      0.539158  ...   
3  57029.000000  57357.000000  57702.000000  58044.000000  58377.000000  ...   
4      0.580334      0.573498      0.599694      0.590951      0.572063  ...   

            2013           2014           2015           2016           2017  \
0      42.990000      43.041000      43.108000      43.192000      43.293000   
1   44351.000000   44666.000000   44978.000000   45293.000000   45614.000000   
2       0.660561       0.707733       0.696090       0.697901       0.706219   
3  103165.000000  103776.000000  104339.000000  104865.000000  105361.000000   
4       0.583290       0.590508       0.541048       0.502860       0.471874   

            2018           2019           2020  2021  Unnamed: 66  
0      43.411000      43.546000      43.697000   NaN          NaN  
1   45949.000000   46294.000000   46654.000000   NaN          NaN  
2       0.731740       0.748028       0.774631   NaN          NaN  
3  105846.000000  106310.000000  106766.000000   NaN          NaN  
4       0.459266       0.437415       0.428017   NaN          NaN  

[5 rows x 64 columns]

In [5]:
data2.head()

Country Name    Afghanistan  Africa Eastern and Southern  \
Year                                                       
1960          886697.085075                 1.669288e+07   
1961          608517.513635                 1.064688e+07   
1962          587279.129589                 1.093390e+07   
1963          635805.437368                 1.123444e+07   
1964          650589.087537                 1.154784e+07   

Country Name  Africa Western and Central        Albania       Algeria  \
Year                                                                    
1960                        1.105452e+07  233877.600823  1.312511e+06   
1961                        8.303902e+06  145931.956281  9.059967e+05   
1962                        8.505636e+06  141214.575494  8.852658e+05   
1963                        8.136713e+06  155071.550760  9.701949e+05   
1964                        8.342463e+06  159657.061949  1.005091e+06   

Country Name  American Samoa      Andorra         Angola  Antigua and Barbuda  \
Year                                                                            
1960            11173.070333  7102.150000  602473.450054          7586.162600   
1961             4307.247758  2942.826521  418861.906797          5125.903503   
1962             4452.354319  3192.169693  400704.632830          4923.926802   
1963             4627.795571  3451.873593  432112.555251          5238.386480   
1964             4811.857565  3722.580122  437573.590456          5300.514789   

Country Name    Arab World  ...    Uzbekistan       Vanuatu  Venezuela, RB  \
Year                        ...                                              
1960          1.513026e+07  ...  2.855853e+06  17619.376000   1.198453e+06   
1961          9.157051e+06  ...  1.971880e+06   7437.914286   8.212302e+05   
1962          9.455672e+06  ...  2.042981e+06   4644.336157   8.082164e+05   
1963          9.767525e+06  ...  2.117773e+06   4966.819799   8.914701e+05   
1964          1.009286e+07  ...  2.197050e+06   5125.917183   9.278776e+05   

Country Name       Vietnam  Virgin Islands (U.S.)  West Bank and Gaza  \
Year                                                                    
1960          3.747841e+06           16970.492333           44.044000   
1961          2.424834e+06            6797.972987          945.535659   
1962          2.358125e+06            6995.668470          837.013928   
1963          2.587906e+06            8025.715674          946.036409   
1964          2.515351e+06            8295.996507          943.765131   

Country Name         World    Yemen, Rep.         Zambia       Zimbabwe  
Year                                                                     
1960          5.916945e+08  527240.620597  604701.677859  850602.614807  
1961          4.218974e+08  361057.279715  329847.471617  410607.347778  
1962          4.471300e+08  347048.088679  315496.766485  390007.171437  
1963          4.841040e+08  374086.558649  354255.160472  441699.004921  
1964          5.059217e+08  380913.965108  245810.369426  297052.596544  

[5 rows x 265 columns]

In [6]:
country_list = pd.read_csv('all_countries.csv')
country_list.head()

name alpha-2 alpha-3  country-code     iso_3166-2   region  \
0     Afghanistan      AF     AFG             4  ISO 3166-2:AF     Asia   
1   Åland Islands      AX     ALA           248  ISO 3166-2:AX   Europe   
2         Albania      AL     ALB             8  ISO 3166-2:AL   Europe   
3         Algeria      DZ     DZA            12  ISO 3166-2:DZ   Africa   
4  American Samoa      AS     ASM            16  ISO 3166-2:AS  Oceania   

        sub-region intermediate-region  region-code  sub-region-code  \
0    Southern Asia                 NaN        142.0             34.0   
1  Northern Europe                 NaN        150.0            154.0   
2  Southern Europe                 NaN        150.0             39.0   
3  Northern Africa                 NaN          2.0             15.0   
4        Polynesia                 NaN          9.0             61.0   

   intermediate-region-code  
0                       NaN  
1                       NaN  
2                       NaN  
3                       NaN  
4                       NaN

In [7]:
data['Country Name'].nunique()

266

In [8]:
country_list['name'].nunique()

249

In [9]:
country_list  = country_list.rename(columns={'name':'Country Name', 'alpha-3':'Country Code', 'region':'Region', 'sub-region':'Sub Region'}, inplace=False)
country_list = country_list[['Country Name', 'Country Code', 'Region', 'Sub Region']]
country_list.head()

Country Name Country Code   Region       Sub Region
0     Afghanistan          AFG     Asia    Southern Asia
1   Åland Islands          ALA   Europe  Northern Europe
2         Albania          ALB   Europe  Southern Europe
3         Algeria          DZA   Africa  Northern Africa
4  American Samoa          ASM  Oceania        Polynesia

In [10]:
df_data = country_list.merge(data, on=['Country Name', 'Country Code']) # Inner join both dataframes
df_data.head()

Country Name Country Code Region     Sub Region  \
0  Afghanistan          AFG   Asia  Southern Asia   
1  Afghanistan          AFG   Asia  Southern Asia   
2  Afghanistan          AFG   Asia  Southern Asia   
3  Afghanistan          AFG   Asia  Southern Asia   
4  Afghanistan          AFG   Asia  Southern Asia   

                             Indicator Name     Indicator Code         1960  \
0  Urban population (% of total population)  SP.URB.TOTL.IN.ZS        8.401   
1                          Urban population        SP.URB.TOTL   755835.000   
2        Urban population growth (annual %)        SP.URB.GROW          NaN   
3                         Population, total        SP.POP.TOTL  8996967.000   
4              Population growth (annual %)        SP.POP.GROW          NaN   

           1961          1962          1963  ...          2013          2014  \
0  8.684000e+00  8.976000e+00  9.276000e+00  ...  2.437300e+01  2.458700e+01   
1  7.962710e+05  8.393850e+05  8.852270e+05  ...  7.865068e+06  8.204880e+06   
2  5.211648e+00  5.272990e+00  5.317463e+00  ...  4.372354e+00  4.229792e+00   
3  9.169406e+06  9.351442e+06  9.543200e+06  ...  3.226959e+07  3.337080e+07   
4  1.898499e+00  1.965805e+00  2.029830e+00  ...  3.494592e+00  3.355602e+00   

           2015          2016          2017          2018          2019  \
0  2.480300e+01  2.502000e+01  2.525000e+01  2.549500e+01  2.575400e+01   
1  8.535606e+06  8.852834e+06  9.164768e+06  9.476982e+06  9.797274e+06   
2  3.951726e+00  3.649128e+00  3.462893e+00  3.349934e+00  3.323827e+00   
3  3.441360e+07  3.538303e+07  3.629611e+07  3.717192e+07  3.804176e+07   
4  3.077054e+00  2.778035e+00  2.547833e+00  2.384309e+00  2.313073e+00   

           2020  2021  Unnamed: 66  
0  2.602600e+01   NaN          NaN  
1  1.013149e+07   NaN          NaN  
2  3.354421e+00   NaN          NaN  
3  3.892834e+07   NaN          NaN  
4  2.303812e+00   NaN          NaN  

[5 rows x 69 columns]

In [11]:
df_data['Country Name'].nunique()

184

Upon merging, there are 184 countries left in the newly created dataframe.

In [12]:
df_data.shape

(13984, 69)

In [13]:
# Melt the dataframe to convert the year columns into a single column

df_data_melted = df_data.melt(id_vars=['Country Name', 'Country Code', 'Region', 'Sub Region', 'Indicator Name', 'Indicator Code'], var_name='Year', value_name='Value')
df_data_melted.head()

Country Name Country Code Region     Sub Region  \
0  Afghanistan          AFG   Asia  Southern Asia   
1  Afghanistan          AFG   Asia  Southern Asia   
2  Afghanistan          AFG   Asia  Southern Asia   
3  Afghanistan          AFG   Asia  Southern Asia   
4  Afghanistan          AFG   Asia  Southern Asia   

                             Indicator Name     Indicator Code  Year  \
0  Urban population (% of total population)  SP.URB.TOTL.IN.ZS  1960   
1                          Urban population        SP.URB.TOTL  1960   
2        Urban population growth (annual %)        SP.URB.GROW  1960   
3                         Population, total        SP.POP.TOTL  1960   
4              Population growth (annual %)        SP.POP.GROW  1960   

         Value  
0        8.401  
1   755835.000  
2          NaN  
3  8996967.000  
4          NaN

In [14]:
# Extracting only the needed indicators from the dataframe.
# Here, Nitrous Oxide Emissios, Methane Emission, CO2 Emission, and Total Population were selected

df_data_1 = df_data_melted[(df_data_melted['Indicator Name'] == "Nitrous oxide emissions (thousand metric tons of CO2 equivalent)")]
df_data_2 = df_data_melted[(df_data_melted['Indicator Name'] == "Methane emissions (kt of CO2 equivalent)")]
df_data_3 = df_data_melted[(df_data_melted['Indicator Name'] == "CO2 emissions (kt)")]
df_data_4 = df_data_melted[(df_data_melted['Indicator Name'] == "Population, total")]

# Check the first five records
df_data_1.head()

Country Name Country Code   Region       Sub Region  \
29      Afghanistan          AFG     Asia    Southern Asia   
105         Albania          ALB   Europe  Southern Europe   
181         Algeria          DZA   Africa  Northern Africa   
257  American Samoa          ASM  Oceania        Polynesia   
333         Andorra          AND   Europe  Southern Europe   

                                        Indicator Name     Indicator Code  \
29   Nitrous oxide emissions (thousand metric tons ...  EN.ATM.NOXE.KT.CE   
105  Nitrous oxide emissions (thousand metric tons ...  EN.ATM.NOXE.KT.CE   
181  Nitrous oxide emissions (thousand metric tons ...  EN.ATM.NOXE.KT.CE   
257  Nitrous oxide emissions (thousand metric tons ...  EN.ATM.NOXE.KT.CE   
333  Nitrous oxide emissions (thousand metric tons ...  EN.ATM.NOXE.KT.CE   

     Year  Value  
29   1960    NaN  
105  1960    NaN  
181  1960    NaN  
257  1960    NaN  
333  1960    NaN

In [15]:
# Rename columns

df_data_1 = df_data_1.rename(columns={'Value':'Nitrous oxide emissions (thousand metric tons of CO2 equivalent)'}, inplace=False)
df_data_1 = df_data_1.drop(['Indicator Name', 'Indicator Code'], axis=1, inplace=False)
df_data_1.head()

Country Name Country Code   Region       Sub Region  Year  \
29      Afghanistan          AFG     Asia    Southern Asia  1960   
105         Albania          ALB   Europe  Southern Europe  1960   
181         Algeria          DZA   Africa  Northern Africa  1960   
257  American Samoa          ASM  Oceania        Polynesia  1960   
333         Andorra          AND   Europe  Southern Europe  1960   

     Nitrous oxide emissions (thousand metric tons of CO2 equivalent)  
29                                                 NaN                 
105                                                NaN                 
181                                                NaN                 
257                                                NaN                 
333                                                NaN

In [16]:
# Rename columns

df_data_2 = df_data_2.rename(columns={'Value':'Methane emissions (kt of CO2 equivalent)'}, inplace=False)
df_data_2 = df_data_2.drop(['Indicator Name', 'Indicator Code'], axis=1, inplace=False)
df_data_2.head()

Country Name Country Code   Region       Sub Region  Year  \
31      Afghanistan          AFG     Asia    Southern Asia  1960   
107         Albania          ALB   Europe  Southern Europe  1960   
183         Algeria          DZA   Africa  Northern Africa  1960   
259  American Samoa          ASM  Oceania        Polynesia  1960   
335         Andorra          AND   Europe  Southern Europe  1960   

     Methane emissions (kt of CO2 equivalent)  
31                                        NaN  
107                                       NaN  
183                                       NaN  
259                                       NaN  
335                                       NaN

In [17]:
# Rename columns

df_data_3 = df_data_3.rename(columns={'Value':"CO2 emissions (kt)"}, inplace=False)
df_data_3 = df_data_3.drop(['Indicator Name', 'Indicator Code'], axis=1, inplace=False)
df_data_3.head()

Country Name Country Code   Region       Sub Region  Year  \
44      Afghanistan          AFG     Asia    Southern Asia  1960   
120         Albania          ALB   Europe  Southern Europe  1960   
196         Algeria          DZA   Africa  Northern Africa  1960   
272  American Samoa          ASM  Oceania        Polynesia  1960   
348         Andorra          AND   Europe  Southern Europe  1960   

     CO2 emissions (kt)  
44                  NaN  
120                 NaN  
196                 NaN  
272                 NaN  
348                 NaN

In [18]:
# Rename columns

df_data_4 = df_data_4.rename(columns={'Value':"Population, total"}, inplace=False)
df_data_4 = df_data_4.drop(['Indicator Name', 'Indicator Code'], axis=1, inplace=False)
df_data_4.head()

Country Name Country Code   Region       Sub Region  Year  \
3       Afghanistan          AFG     Asia    Southern Asia  1960   
79          Albania          ALB   Europe  Southern Europe  1960   
155         Algeria          DZA   Africa  Northern Africa  1960   
231  American Samoa          ASM  Oceania        Polynesia  1960   
307         Andorra          AND   Europe  Southern Europe  1960   

     Population, total  
3            8996967.0  
79           1608800.0  
155         11057864.0  
231            20127.0  
307            13410.0

In [19]:
# Merge the respective dataframes

df_merged = df_data_1.merge(df_data_2, on=['Country Name', 'Country Code', 'Region', 'Sub Region', 'Year'])
df_merged = df_merged.merge(df_data_3, on=['Country Name', 'Country Code', 'Region', 'Sub Region', 'Year'])
df_merged = df_merged.merge(df_data_4, on=['Country Name', 'Country Code', 'Region', 'Sub Region', 'Year'])
df_merged.head()

Country Name Country Code   Region       Sub Region  Year  \
0     Afghanistan          AFG     Asia    Southern Asia  1960   
1         Albania          ALB   Europe  Southern Europe  1960   
2         Algeria          DZA   Africa  Northern Africa  1960   
3  American Samoa          ASM  Oceania        Polynesia  1960   
4         Andorra          AND   Europe  Southern Europe  1960   

   Nitrous oxide emissions (thousand metric tons of CO2 equivalent)  \
0                                                NaN                  
1                                                NaN                  
2                                                NaN                  
3                                                NaN                  
4                                                NaN                  

   Methane emissions (kt of CO2 equivalent)  CO2 emissions (kt)  \
0                                       NaN                 NaN   
1                                       NaN                 NaN   
2                                       NaN                 NaN   
3                                       NaN                 NaN   
4                                       NaN                 NaN   

   Population, total  
0          8996967.0  
1          1608800.0  
2         11057864.0  
3            20127.0  
4            13410.0

In [20]:
# Rename columns in the merged dataframe

df_merged = df_merged.rename(columns={'Nitrous oxide emissions (thousand metric tons of CO2 equivalent)':'NO Emission', 'Methane emissions (kt of CO2 equivalent)':'Methane Emission', 'CO2 emissions (kt)':'CO2 Emission', 'Population, total':'Total Population'}, inplace=False)
df_merged.head()

Country Name Country Code   Region       Sub Region  Year  NO Emission  \
0     Afghanistan          AFG     Asia    Southern Asia  1960          NaN   
1         Albania          ALB   Europe  Southern Europe  1960          NaN   
2         Algeria          DZA   Africa  Northern Africa  1960          NaN   
3  American Samoa          ASM  Oceania        Polynesia  1960          NaN   
4         Andorra          AND   Europe  Southern Europe  1960          NaN   

   Methane Emission  CO2 Emission  Total Population  
0               NaN           NaN         8996967.0  
1               NaN           NaN         1608800.0  
2               NaN           NaN        11057864.0  
3               NaN           NaN           20127.0  
4               NaN           NaN           13410.0

In [21]:
# Extract data from year 2000 and above

final_data = df_merged[df_merged['Year']>='2000']
final_data.head()

Country Name Country Code   Region       Sub Region  Year  \
7360     Afghanistan          AFG     Asia    Southern Asia  2000   
7361         Albania          ALB   Europe  Southern Europe  2000   
7362         Algeria          DZA   Africa  Northern Africa  2000   
7363  American Samoa          ASM  Oceania        Polynesia  2000   
7364         Andorra          AND   Europe  Southern Europe  2000   

      NO Emission  Methane Emission  CO2 Emission  Total Population  
7360       3620.0           10590.0         770.0        20779957.0  
7361       1180.0            3750.0        3170.0         3089027.0  
7362       7400.0           39920.0       78590.0        31042238.0  
7363          NaN               NaN           NaN           57816.0  
7364          0.0              30.0         520.0           65390.0

In [22]:
# Check for the number of rows and columns

final_data.shape

(4232, 9)

In [23]:
# Check the number of records from the respective geographical regions

final_data['Region'].value_counts()

Africa      1104
Europe       966
Asia         920
Americas     828
Oceania      414
Name: Region, dtype: int64

In [24]:
# Check for missing observations

inal_data.isnull().mean()

NameError: name 'inal_data' is not defined

In [ ]:
# Drop missing observations

final_data = final_data.dropna(subset=['NO Emission'])
final_data.head()

In [ ]:
# Check for the percentage of missing observations

final_data.isnull().mean()

In [ ]:
# Drop missing observations

final_data = final_data.dropna(subset=['Total Population'])
final_data.head()

In [ ]:
# Check for missing observations

final_data.isnull().mean()

In [ ]:
# Check data info

final_data.info()

In [ ]:
# Convert columns to float data type

final_data['NO Emission'] = pd.to_numeric(final_data['NO Emission'], errors='coerce')
final_data['Methane Emission'] = pd.to_numeric(final_data['Methane Emission'], errors='coerce')
final_data['CO2 Emission'] = pd.to_numeric(final_data['CO2 Emission'], errors='coerce')
final_data['Total Population'] = pd.to_numeric(final_data['Total Population'], errors='coerce')

final_data.info()

In [ ]:
# Correlation analysis for the selected indicators

corr = final_data.set_index('Year').corr()
corr

In [ ]:
# Regression Heatmap

sns.heatmap(corr)

From the above correlation analysis, we found that there exists strong positive corellation between the variables.

[Interpretation]
1. There will an increment by in Nitrous Oxide emission per unit increase in population by about 92% provided that other variables remained constant.

2. There will an increment by in Methan emission per unit increase in population by 84% provided that other variables remained constant.

3. There will an increment by in CO2 emission per unit increase in population by 81% provided that other variables remained constant.


From the above result, we are safe to safe that the emission of thee three gases (Nitrous Oxide, Methane and CO2), had great impact on population size and vice versa.

Here we shall be focusing on countries from the African continent (being the region with the most records). We will create a new dataframe for records coming only from Africa.

In [ ]:
# Create a new dataframe for only the African region

df_africa = final_data[final_data['Region']=='Africa']
df_africa.head()

In [ ]:
# Grouping the data by year. Hence, the average yearly average of the respective indicators will be calculated.

df_africa_year_grouped = df_africa[['Year', 'NO Emission', 'Methane Emission', 'CO2 Emission', 'Total Population']].groupby('Year').agg('mean')
df_africa_year_grouped = df_africa_year_grouped.reset_index()
df_africa_year_grouped.head()

In [ ]:
# Summary Statistics

df_africa_year_grouped.describe()

In Africa, over the 19 years period under study, the minimum, maximum and average values of the respective indicators are:

1. Nitrous Oxide: 7649.3750, 10430.4255 and 9030.6849 respectively.

2. Methane: 14168.5417, 20372.1277, and 17758.0517 respectively.

3. CO2: 13172.7083, 2332.7659, and 18717.8782 respectively.

4. Population: 1.3322e+07, 2.1263e+07 and 1.6945e+07 respectively.

In [ ]:
df_africa_year_grouped.plot(x="Year", y=['NO Emission', 'Methane Emission', 'CO2 Emission',], kind="bar", figsize=(10,8), title='Bar Plot of Gaseous Emissions')
plt.legend()
plt.show()

In Africa, gaseous emission has maintained a steady rise over the years, leaving none of the three gases out. This could be an indication that there are a lot of industrialization in Africa over the years.

In [ ]:
df_africa_year_grouped.plot(x='Year', y='CO2 Emission', kind='scatter', figsize=(10, 5), title='Scatter Plot of CO2 Emission')

plt.show()

In [ ]:
df_africa_year_grouped.plot(x='Total Population', y='CO2 Emission', kind='scatter', figsize=(10, 5), title='Scatter Plot of Population vs CO2 Emission')

plt.show()

In [ ]:
df_africa_year_grouped.plot(x='Total Population', y='Methane Emission', kind='scatter', figsize=(10, 5), title='Scatter Plot of Population vs Methane Emission')

plt.show()

In [ ]:
df_africa_year_grouped.plot(x='Total Population', y='NO Emission', kind='scatter', figsize=(10, 5), title='Scatter Plot of Population vs Nitrous Oxide Emission')

plt.show()

The above scatter plots further support the initial claims that in Africa, gaseous emission has been on a steady over the years.

CO2 Emission is a major contributor to ozone layer depletion, and hence a great threat to climate sustainability. While we have been able to establish that there is increased gaseous emissions in Africa over the years, then, Africa might be a great risk of adverse/severe climatic actions over time.

Having identified that CO2 emission had the highest value out of all 3 gases being compared, we shall look at countries in europe with highest records of CO2 emission

In [ ]:
# Grouping the data by country

df_africa_country_grouped = df_africa[['Country Name', 'Year', 'NO Emission', 'Methane Emission', 'CO2 Emission', 'Total Population']].groupby('Country Name').agg('mean').sort_values(by='CO2 Emission', ascending=True)
df_africa_country_grouped = df_africa_country_grouped.reset_index()
df_africa_country_grouped.head()

In [ ]:
# Summary Statistics

df_africa_country_grouped.describe()

The minimum, maximum, and average values for the respective gaseous emissions are:

1. Nitrous Oxide: 0.0000, 53677.3684, and 8989.7013

2. Methane: 20.00, 778381.0526, and 36172.4324

3. CO2: 1.9579e+02, 1.5655e+06, and 1.3808e+05

In [ ]:
# Checking for the number of rows and columns
df_africa_year_grouped.shape

In [ ]:
# Top 10 countries with CO2 Emission

top_10_countries_co2 = df_africa_country_grouped.nlargest(10, 'CO2 Emission')
top_10_countries_co2.head(10)

South Africa took the lead in CO2 Emission, while Sao Tome and Principe had the lowest record of CO2 Emission. We shall then take a look at the record of both South Africa and Sao Tome and Principe respectively.

### South Africa

In [ ]:
# Create a dataframe for South Africa

df_SA = final_data[final_data['Country Name']=='South Africa']
df_SA.head()

In [ ]:
# Descriptive Statistics

df_SA.describe()

In South Africa, over the years, the the average value for the selected indicators was:

1. Nitrous Oxide: 17510.0000, 20020.0000, and 19053.1579 respectively.

2. Methane:34040.0000, 45140.0000, and 41385.2632 respectively.

3. CO2: 284730.0000, 447980.0000, and 395723.6842 respectively.

4. Total Population: 4.4968e+07, 5.7793e+07, and 5.0901e+07 respectively.

In [ ]:
# Bar Plot of Gaseous Emission in South Africa

df_SA.plot(x="Year", y=['NO Emission', 'Methane Emission', 'CO2 Emission',], kind="bar", figsize=(10,8), title='Bar Plot of Gaseous Emissions in South Africa')

plt.legend()
plt.show()

From the above chart, we found that CO2 emission accounts for the largest records of gaseous emissions in South Africa.

In [ ]:
# Creating another dataframe for Sao Tome and Prinipe

df_ST = final_data[final_data['Country Name']=='Sao Tome and Principe']
df_ST.head()

In [ ]:
# Summary Statistics

df_ST.describe()

In South Africa, over the years, the the average value for the selected indicators was:

1. Nitrous Oxide: 10.0000, 20.0000, and 17.3684 respectively.

2. Methane: 20.0000, 40.0000, and 25.7895 respectively.

3. CO2: 50.0000, 140.0000, and 94.2105 respectively.

4. Total Population: 142264.0000, 211032.0000, and 175417.2105 respectively.

In [ ]:
# Bar Plot of Gaseous Emissions in Sao Tome and Principe

df_ST.plot(x="Year", y=['NO Emission', 'Methane Emission', 'CO2 Emission',], kind="bar", figsize=(10,8), title='Bar Plot of Gaseous Emissions in Sao Tome and Principe')

plt.legend()
plt.show()

From the above chart, we found that CO2 emission accounts for the largest records of gaseous emissions in Sao Tome and Principe.

## Comparison of the findings from Africa as whole, South Africa & Sao Tome and Prince

Unarguably, we were able to establish that in Africa, gaseous emission is fast increasing, and might be a cause for alarm for the African state. Quite worrisome is the fact that the gas with the most emission happened to be CO2 which is a major concern. On a norms, while the average CO2 emission in Africa was: 18717.8782, in South Africa; 395723.684211, while it was 94.2105 respectively.

The above also shows that South Africa had more prevalence of as gaseous emtssion than the two other gases.